src/Integration.UnitTests/Suppression/SuppressionIssueMatcherTests.cs

```diff
@@ -0,0 +1,145 @@
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration.Suppression;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Suppression
+{
+    [TestClass]
+    public class SuppressionIssueMatcherTests
+    {
+        private SuppressionIssueMatcher testSubject;
+        private Mock<ISonarQubeIssuesProvider> mockIssuesProvider;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            mockIssuesProvider = new Mock<ISonarQubeIssuesProvider>();
+            testSubject = new SuppressionIssueMatcher(mockIssuesProvider.Object);
+        }
+
+        [TestMethod]
+        public void Ctor_NullProvider_Throws()
+        {
+            Action act = () => new SuppressionIssueMatcher(null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("issuesProvider");
+        }
+
+        [TestMethod]
+        public void MatchExists_NullIssue_Throws()
+        {
+            Action act = () => testSubject.SuppressionExists(null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("issue");
+        }
+
+        [TestMethod]
+        public void MatchExists_NoServerIssues_ReturnsFalse()
+        {
+            // Arrange
+            var issueToMatch = CreateIssueToMatch("rule1", 1, "hash1");
+            ConfigureServerIssues(issueToMatch /* no server issues */);
+
+            // Act and assert
+            testSubject.SuppressionExists(issueToMatch).Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void MatchExists_NoMatches_ReturnsFalse()
+        {
+            // Arrange
+            var issueToMatch = CreateIssueToMatch("RightRuleId", 1, "RightHash");
+
+            ConfigureServerIssues(issueToMatch,
+                CreateServerIssue("WrongRuleId", 1, "RightHash"),  // wrong rule id
+                CreateServerIssue("RightRuleId", 2, "wrong hash"), // wrong line and hash
+                CreateServerIssue("RightRuleId", 3, "RIGHTHASH")); // wrong hash and wrong-case hash
+
+            // Act and assert
+            testSubject.SuppressionExists(issueToMatch).Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void MatchExists_MatchingIdAndLine_ReturnsTrue()
+        {
+            // Arrange
+            var issueToMatch = CreateIssueToMatch("RightRuleId", 1, "RightHash");
+            ConfigureServerIssues(issueToMatch,
+                CreateServerIssue("wrong rule ID", 2, "RightHash"),
+                CreateServerIssue("RIGHTRULEID", 1, "WrongHash") // rule id comparison is case-insensitive -> match on line
+                );
+
+            // Act and assert
+            testSubject.SuppressionExists(issueToMatch).Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void MatchExists_MatchingIdAndLineHash_ReturnsTrue()
```

Comment:
I think it's worth separating the test cases into different test methods, for better readability. Every case here deserves its own test. It might also be worth doing DataTestMethod.
So something like 
MatchExists(string serverIssueRuleId, int serverIssueLineNumber, string serverIssueHash, bool shouldMatch)
[DataRow("right rule id", 1, "right hash", true)]
[DataRow("right rule id", 2, "right hash", true)]
[DataRow("right rule id", 1, "wrong hash", false)]